src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/ChunkedEncodingReadStream.cs

```diff
@@ -27,8 +28,13 @@ private sealed class ChunkedEncodingReadStream : HttpContentReadStream
             private ulong _chunkBytesRemaining;
             /// <summary>The current state of the parsing state machine for the chunked response.</summary>
             private ParsingState _state = ParsingState.ExpectChunkHeader;
+            private HttpResponseMessage _response;
```

Comment:
The connection already has a reference to the response object (I think?), so I don't think we need this. My suggestion would be to refactor this so that we have a method on the connection like ParseTrailingHeaders, and call into it from this code. That has the additional advantage that it's easy to kill the connection on parse errors.

Reply:
> The connection already has a reference to the response object (I think?)

Right, let me think how to refactor this.

Reply:
Actually no, the response object only is created/exists in HttpConnection.SendAsyncCore() method, Connection itself doesn't has a reference to the response.

Regarding "it's easy to kill the connection on parse errors", in `ChunkedEncodingReadStream` we will dispose the connection and set it to null upon getting Exception.